test/ModuleInterface/back-deploy-attr.swift

```diff
@@ -17,27 +17,24 @@
 
 public struct TopLevelStruct {
   // CHECK: @_backDeploy(macOS 11.0)
-  // CHECK: public func backDeployedFunc1_SinglePlatform() -> Swift.Int
+  // FROMSOURCE: public func backDeployedFunc1_SinglePlatform() -> Swift.Int { return 42 }
+  // FROMMODULE: public func backDeployedFunc1_SinglePlatform() -> Swift.Int
   @available(macOS 12.0, *)
   @_backDeploy(macOS 11.0)
-  public func backDeployedFunc1_SinglePlatform() -> Int {
-    return 42
-  }
+  public func backDeployedFunc1_SinglePlatform() -> Int { return 42 }
   
   // CHECK: @_backDeploy(macOS 11.0)
   // CHECK: @_backDeploy(iOS 14.0)
-  // CHECK: public func backDeployedFunc2_MultiPlatform() -> Swift.Int
+  // FROMSOURCE: public func backDeployedFunc2_MultiPlatform() -> Swift.Int { return 43 }
+  // FROMMODULE: public func backDeployedFunc2_MultiPlatform() -> Swift.Int
   @available(macOS 12.0, *)
   @_backDeploy(macOS 11.0, iOS 14.0)
```

Comment:
This is probably from the previous PR but I just noticed it. The logic here uses `@available` to describe the OS version with the function in the dylib and ` @_backDeploy` points to older OS versions where we can emit the code in the clients, is that right?

Is there a specific motivation for this order? I'm wondering if having `@available` pointing to the older OS where this is first available with no concern for back-deployment, wouldn't make it easier to support existing availability checking. Then we'd need to concern ourselves with the ` @_backDeploy`'s version only for inlining and linking.

Reply:
Yes, your description of how they are intended to interact is right. There were a couple reasons I went with this design:
- It makes it very simple to add back deployment to an existing declaration since you only need to add the new attribute and fill it in.
- It keeps the effect of `@available` on ABI unchanged since the attributed declaration continues to only indicate that the declaration became ABI starting in the introduced OS versions.

I was thinking of this as the least disruptive implementation and wanted to wait for community feedback on the design before refining it, but maybe I am underestimating how much would need to change in availability checking to ensure the current design works.